cmd/cluster-agent/app/app.go

```diff
@@ -227,10 +233,9 @@ func start(cmd *cobra.Command, args []string) error {
 
 	// Cancel the main context to stop components
 	mainCtxCancel()
+	// wait for the External Metrics Server to stop properly
+	wg.Wait()
```

Comment:
why do we use a `WaitGroup` for a single go routine?

Reply:
The Run server is a blocking call, it's cleaner to propagate  context cancel everywhere and then clone the server IMO

Reply:
I mean we could've used a `done` chan instead of the waitgroup, but this work fine as well 👍